src/main/kotlin/no/nav/medlemskap/services/udi/UdiMapper.kt

```diff
@@ -1,21 +1,140 @@
+            ArbeidOmfangKategori.KUN_ARBEID_HELTID -> Arbeidsomfang.KUN_ARBEID_HELTID
+            ArbeidOmfangKategori.KUN_ARBEID_DELTID -> Arbeidsomfang.KUN_ARBEID_DELTID
+            ArbeidOmfangKategori.DELTID_SAMT_FERIER_HELTID -> Arbeidsomfang.DELTID_SAMT_FERIER_HELTID
+            ArbeidOmfangKategori.INGEN_KRAV_TIL_STILLINGSPROSENT -> Arbeidsomfang.INGEN_KRAV_TIL_STILLINGSPROSENT
+            ArbeidOmfangKategori.UAVKLART -> Arbeidsomfang.UAVKLART
+        }
+    private fun mapArbeidsadgangType(arbeidsadgang: Arbeidsadgang) =
+        when (arbeidsadgang.typeArbeidsadgang) {
+            no.udi.mt_1067_nav_data.v1.ArbeidsadgangType.BESTEMT_ARBEIDSGIVER_ELLER_OPPDRAGSGIVER -> ArbeidsAdgangType.BESTEMT_ARBEIDSGIVER_ELLER_OPPDRAGSGIVER
+            no.udi.mt_1067_nav_data.v1.ArbeidsadgangType.BESTEMT_ARBEID_ELLER_OPPDRAG -> ArbeidsAdgangType.BESTEMT_ARBEID_ELLER_OPPDRAG
+            no.udi.mt_1067_nav_data.v1.ArbeidsadgangType.GENERELL -> ArbeidsAdgangType.GENERELL
+            no.udi.mt_1067_nav_data.v1.ArbeidsadgangType.UAVKLART -> ArbeidsAdgangType.UAVKLART
+            no.udi.mt_1067_nav_data.v1.ArbeidsadgangType.BESTEMT_ARBEIDSGIVER_OG_ARBEID_ELLER_BESTEMT_OPPDRAGSGIVER_OG_OPPDRAG -> ArbeidsAdgangType.BESTEMT_ARBEIDSGIVER_OG_ARBEID_ELLER_BESTEMT_OPPDRAGSGIVER_OG_OPPDRAG
+        }
+
+    private fun mapPeriode(arbeidsadgang: Arbeidsadgang): Periode {
+
+        return Periode(
+            fom = arbeidsadgang.arbeidsadgangsPeriode.fra.asLocalDate(),
+            tom = arbeidsadgang.arbeidsadgangsPeriode.til.asLocalDate()
+
+        )
+    }
+
+    private fun mapGjeldendeOppholdsstatus(gjeldendeOppholdsstatus: GjeldendeOppholdsstatus): Boolean {
+        if (gjeldendeOppholdsstatus.eoSellerEFTAOpphold != null) {
+            return false
+            // return mapEosEllerEftaopphold(gjeldendeOppholdsstatus.eoSellerEFTAOpphold)
+        }
+        if (gjeldendeOppholdsstatus.oppholdstillatelseEllerOppholdsPaSammeVilkar != null) {
+            return false
+            // return mapOppholdstillatelseEllerOppholdsPaSammeVilkar(gjeldendeOppholdsstatus.oppholdstillatelseEllerOppholdsPaSammeVilkar)
+        }
+        if (gjeldendeOppholdsstatus.ikkeOppholdstillatelseIkkeOppholdsPaSammeVilkarIkkeVisum != null) {
+            return false
+            // return mapIkkeOppholdstillatelseIkkeOppholdsPaSammeVilkarIkkeVisum(gjeldendeOppholdsstatus.ikkeOppholdstillatelseIkkeOppholdsPaSammeVilkarIkkeVisum)
+        }
+        if (gjeldendeOppholdsstatus.uavklart != null) {
+            return false
+        }
+        return false
+    }
+
+    private fun mapIkkeOppholdstillatelseIkkeOppholdsPaSammeVilkarIkkeVisum(ikkeOppholdstillatelseIkkeOppholdsPaSammeVilkarIkkeVisum: IkkeOppholdstillatelseIkkeOppholdsPaSammeVilkarIkkeVisum?) {
+        TODO("Not yet implemented")
+    }
+
+    private fun mapOppholdstillatelseEllerOppholdsPaSammeVilkar(oppholdstillatelseEllerOppholdsPaSammeVilkar: OppholdstillatelseEllerOppholdsPaSammeVilkar?): Boolean {
```

Comment:
Ser UDI har et dårlig navn på det med oppholdstillatelseEllerOppholds.. Finnes et bedre navn for oss som blir mer forståelig?